components/mpas-ocean/src/shared/mpas_ocn_tracer_ideal_age.F

```diff
@@ -145,20 +150,48 @@ end subroutine ocn_tracer_ideal_age_compute!}}}
 !  routine ocn_tracer_ideal_age_init
 !
 !> \brief   Initializes ocean ideal age
-!> \author  Todd Ringler
-!> \date    06/09/2015
+!> \author  Mathew Maltrud
+!> \date    08/19/2021
 !> \details
 !>  This routine initializes fields required for tracer ideal age
 !
 !-----------------------------------------------------------------------
 
-   subroutine ocn_tracer_ideal_age_init(err)!{{{
+   subroutine ocn_tracer_ideal_age_init(domain,err)!{{{
+
+      type (domain_type), intent(inout) :: domain !< Input/Output: domain information
 
       integer, intent(out) :: err !< Output: error flag
 
+      type (mpas_pool_type), pointer :: forcingPool
+      type (mpas_pool_type), pointer :: tracersIdealAgeFieldsPool
+
+      integer, dimension(:), pointer :: &
+         landIceMask
+
+      real (kind=RKIND), dimension(:,:), pointer :: &
+         idealAgeMask
+
       err = 0
 
-   !--------------------------------------------------------------------
+      call mpas_pool_get_subpool(domain % blocklist % structs, 'forcing', forcingPool)
+      call mpas_pool_get_subpool(forcingPool, &
+                                 'tracersIdealAgeFields', &
+                                  tracersIdealAgeFieldsPool)
+      call mpas_pool_get_array(tracersIdealAgeFieldsPool, &
+                               'idealAgeTracersIdealAgeMask', &
+                                idealAgeMask)
+
+      call mpas_pool_get_array(forcingPool, 'landIceMask', landIceMask)
+
+      if (associated(idealAgeMask)) then
+
+      ! set mask = 0 for open ocean
+      ! set mask = 1 under ice shelves
+      idealAgeMask = 0.0_RKIND
+      where (landIceMask /= 0) idealAgeMask(1,:) = 1.0_RKIND
+
+      endif !  associated(idealAgeMask)
```

Comment:
@maltrud, thanks for updating this. The indentation would need to be fixed.

But more importantly, @philipwjones had asked for a different solution that uses one of the logical variables to determine if `idealAgeMask` would be available rather than `associated()`.  Could you switch to that approach here and in the other similar code blocks instead?